compiler/base.ts

```diff
@@ -323,15 +324,53 @@ export class LineCompiler implements InstrVisitor<void>, ExprVisitor<void> {
     return this.lines[this.currentLine].instructions[this.currentInstrNo];
   }
 
-  private createSyntaxError(instr: Instr, message: string): SyntaxError {
+  private last(): Instr | null {
+    if (typeof this._last !== 'undefined') {
+      return this._last;
+    }
+
+    for (let i = this.lines.length - 1; i >= 0; i--) {
+      const line = this.lines[i];
+      const j = line.instructions.length - 1;
+      if (j >= 0) {
+        this._last = line[j];
+        return line[j];
+      }
+    }
+    this._last = null;
+    return null;
+  }
+
+  private getInstrOffsets(instr: Instr | null): [number, number] {
+    let offsetStart: number;
+    let offsetEnd: number;
+    if (instr) {
+      offsetStart = instr.offsetStart;
+      offsetEnd = instr.offsetEnd;
+    } else {
+      const last = this.last();
+      if (last) {
+        offsetStart = last.offsetEnd;
```

Comment:
This is `offsetEnd` because I'm really trying to point to an EOF, rather than a true Instruction.